### PR TITLE
linting test on GH actions.

### PR DESCRIPTION
**Pull request by nicolas-rabault**
Setting up a clean code linter with a fixed format on projects and automate it.
Actions need to raise an error in case of failure and display errors as diff.
The lining should be callable in the command line locally.

### CODE REVIEW

**Comment by nicolas-rabault** [comment]
This PR could generate a lot of conflicts on our actual dev branch. I suggest to remove the https://github.com/Luos-io/Luos/pull/82/commits/bd892f99d093eaedc290fc38b03e93562532f1c8 commit and rebase this branch on top of the current dev branch just before the merge avoiding any conflict and making everything clean smoothly.

File: Robus/src/msg_alloc.c [hunk 1]
```diff
@@ -562,8 +562,7 @@ static inline error_return_t MsgAlloc_ClearMsgSpace(void *from, void *to)
  ******************************************************************************/
 static inline error_return_t MsgAlloc_CheckMsgSpace(void *from, void *to)
 {
-    if ((((uint32_t)used_msg >= (uint32_t)from) && ((uint32_t)used_msg <= (uint32_t)to)) ||
-        (((uint32_t)oldest_msg >= (uint32_t)from) && ((uint32_t)oldest_msg <= (uint32_t)to)))
+    if ((((uint32_t)used_msg >= (uint32_t)from) && ((uint32_t)used_msg <= (uint32_t)to)) || (((uint32_t)oldest_msg >= (uint32_t)from) && ((uint32_t)oldest_msg <= (uint32_t)to)))
```
Comment (JeromeGalan) [inline Robus/src/msg_alloc.c:565]: Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

File: src/routing_table.c [hunk 2]
```diff
@@ -204,15 +204,7 @@ char *RoutingTB_StringFromType(luos_type_t type)
...
-    if ((type == ANGLE_MOD) ||
-        (type == STATE_MOD) ||
-        (type == DYNAMIXEL_MOD) ||
-        (type == DISTANCE_MOD) ||
-        (type == IMU_MOD) ||
-        (type == LOAD_MOD) ||
-        (type == CONTROLLER_MOTOR_MOD) ||
-        (type == VOLTAGE_MOD) ||
-        (type == LIGHT_MOD))
+    if ((type == ANGLE_MOD) || (type == STATE_MOD) || (type == DYNAMIXEL_MOD) || (type == DISTANCE_MOD) || (type == IMU_MOD) || (type == LOAD_MOD) || (type == CONTROLLER_MOTOR_MOD) || (type == VOLTAGE_MOD) || (type == LIGHT_MOD))
```
Comment (JeromeGalan) [inline src/routing_table.c:207]: Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

File: Robus/src/reception.c [hunk 3]
```diff
@@ -134,8 +134,7 @@ void Recep_GetData(volatile uint8_t *data)
     }
     else if (data_count > data_size)
     {
-        uint16_t crc = ((uint16_t)current_msg->data[data_size]) |
-                       ((uint16_t)current_msg->data[data_size + 1] << 8);
+        uint16_t crc = ((uint16_t)current_msg->data[data_size]) | ((uint16_t)current_msg->data[data_size + 1] << 8);
```
Comment (JeromeGalan) [inline Robus/src/reception.c:137]: Replace | by ||.
Perhaps we should put OR operator || in the second line.  So, after clang format processing, the conditions should be on different lines.

Comment (nicolas-rabault) [reply Robus/src/reception.c:137]: Here is a logical or so we have to keep the |

**Review by JeromeGalan — APPROVED** [review]

**Pull request merged**